src/models/ui.ts

```diff
@@ -0,0 +1,49 @@
+import { types } from "mobx-state-tree";
+
+type ToggleElement = "learningLogExpanded" | "leftNavExpanded" | "myWorkExpanded";
+
+export const UIModel = types
+  .model("UI", {
+    learningLogExpanded: false,
+    leftNavExpanded: false,
+    myWorkExpanded: false,
+  })
+  .actions((self) => {
+    const toggleWithOverride = (toggle: ToggleElement, override?: boolean) => {
+      const expanded = typeof override !== "undefined" ? override : !self[toggle];
+
+      // for mobx we can't set self[toggle] as it doesn't trigger the update
```

Comment:
What does the first part of this comment mean?

Reply:
The way that Mobx wraps properties in getters and setters to that trigger updates in components, `self.learningLogExpanded = true` triggers the update process but `self["learningLogExpanded"] = true` does not which is why I had to add the switch statement.

Reply:
Ahh makes sense, thanks for the explanation

Reply:
According to [What does MobX react to? ](https://github.com/mobxjs/mobx/blob/gh-pages/docs/best/react.md) MobX should handle the bracket access just fine. It seems like there may be something else going on here.